### PR TITLE
fix set_min_numpy_version

### PR DESCRIPTION
**Pull request by mboisson**
This fixes https://github.com/ComputeCanada/wheels_builder/issues/73

### CODE REVIEW

**Review by ccoulombe — APPROVED** [review]
lgtm

File: manipulate_wheels.py [hunk 1]
```diff
@@ -150,12 +150,11 @@ def main():
                         if re.search(r'^numpy(\W|$)', curr_req):
                             if args.verbose:
                                 print('Found numpy dependency.')
-                            # find exiting version requirements (split on '(' and ')')
-                            parse_req = re.search(r"^([^(]+\()([^)]+)(\).*)", curr_req)
-                            if parse_req:
-                                req_tokens = list(parse_req.groups())
+                            curr_version_req = curr_req.replace("numpy","").strip()
+                            req_tokens = ["numpy",""]
```
[the remark this answers is not in the document]
Comment (mboisson) [reply manipulate_wheels.py:154]: Not really. 
`curr_version_req` will contain whatever is after `numpy`, which then gets parsed in following lines. 

`req_tokens` will contain the new requirements, and will be adjusted in the following lines.

[the remark this answers is not in the document]
Comment (mboisson) [reply manipulate_wheels.py:154]: The logic remains the same as before. The only thing which changes is the parsing is no longer made with regex and no longer expects parenthesis.

[the remark this answers is not in the document]
Comment (ostueker) [reply manipulate_wheels.py:154]: I see. There seem to be two possible notations: with and without parenthesis.

But now the logic breaks when a numpy-version is specified with parenthesis:
```console
$ manipulate_wheels_2.py -p -w original_wheels/GromacsWrapper-0.8.2-py2.py3-none-any.whl | head -n4
Resulting wheels will be in directory ./tmp
Requirements for wheel original_wheels/GromacsWrapper-0.8.2-py2.py3-none-any.whl:
==========================
numpy (>=1.0)

$ manipulate_wheels_2.py -w original_wheels/GromacsWrapper-0.8.2-py2.py3-none-any.whl -t cc --set_min_numpy "0.9" 
Resulting wheels will be in directory ./tmp
New wheel created tmp/GromacsWrapper-0.8.2+cc-py2.py3-none-any.whl

$ manipulate_wheels_2.py -p -w tmp/GromacsWrapper-0.8.2+cc-py2.py3-none-any.whl | head -n4
Resulting wheels will be in directory ./tmp
Requirements for wheel tmp/GromacsWrapper-0.8.2+cc-py2.py3-none-any.whl:
==========================
numpy>=0.9
```
i.e. it happily lowered the version below the minimal version the original wheel requested.

The old code did the right thing in that case:
```console
$ manipulate_wheels.py -w original_wheels/GromacsWrapper-0.8.2-py2.py3-none-any.whl -t cc1 --set_min_numpy "0.9" 
Resulting wheels will be in directory ./tmp
New wheel created tmp/GromacsWrapper-0.8.2+cc1-py2.py3-none-any.whl

$ manipulate_wheels.py -p -w tmp/GromacsWrapper-0.8.2+cc1-py2.py3-none-any.whl | head -n4
Resulting wheels will be in directory ./tmp
Requirements for wheel tmp/GromacsWrapper-0.8.2+cc1-py2.py3-none-any.whl:
==========================
numpy (>=1.0)
```

The new code works correctly with the other notation:
```console
$ manipulate_wheels_2.py -p -w original_wheels/scipy-1.10.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl | head -n4
Resulting wheels will be in directory ./tmp     
Requirements for wheel original_wheels/scipy-1.10.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl:
==========================                 
numpy<1.27.0,>=1.19.5                                                                                   

$ manipulate_wheels_2.py -w original_wheels/scipy-1.10.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl  --set_min_numpy "1.18.2" 
Resulting wheels will be in directory ./tmp     
New wheel created tmp/scipy-1.10.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl                                                                                                                   

$ manipulate_wheels_2.py -p -w tmp/scipy-1.10.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl  | head -n4
Resulting wheels will be in directory ./tmp
Requirements for wheel tmp/scipy-1.10.1-cp310-cp310-manylinux_2_17_x86_64.manylinux2014_x86_64.whl:
==========================                      
numpy<1.27.0,>=1.19.5
```

[the remark this answers is not in the document]
Comment (mboisson) [reply manipulate_wheels.py:154]: Ah, good point. I guess stripping the parenthesis would work in all cases ?

[the remark this answers is not in the document]
Comment (mboisson) [reply manipulate_wheels.py:154]: I pushed a commit which removes parentheses

**Comment by ccoulombe** [comment]
`y_py` config already exists in the config directory ?

**Comment by mboisson** [comment]
> `y_py` config already exists in the config directory ?

we added it both at the same time... it made its way to this branch by mistake, but now it's there, so I resolved the conflict.

File: manipulate_wheels.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (mboisson) [reply manipulate_wheels.py:154]: @ostueker I confirmed with your example that the new code (stripping the parentheses) fixes that, and the minimum version of numpy in that example remains >=1.0

**Pull request merged**